src/Services/CereProc.php

```diff
@@ -0,0 +1,201 @@
+     * @var integer
+     */
+    private int $sampleRate = 48000;
+
+    public function __construct(string $audioFormat = 'mp3', int $sampleRate = 48000)
+    {
+        $this->setAudioFormat($audioFormat);
+        $this->setSampleRate($sampleRate);
+    }
+
+    /**
+     * Returns the full name of this service.
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * Returns the short name of this service.
+     *
+     * @return string
+     */
+    public function getShortName(): string
+    {
+        return $this->shortName;
+    }
+
+    /**
+     * Returns the default voice ID to be used with this service.
+     *
+     * @return string
+     */
+    public function getDefaultVoice(): string
+    {
+        return $this->defaultVoice;
+    }
+
+    /**
+     * Sets the format for the audio ouput
+     *
+     * @param string $audioFormat
+     * @return void
+     */
+    public function setAudioFormat(string $audioFormat)
+    {
+        if (!in_array($audioFormat, ['mp3', 'ogg', 'wav'])) $audioFormat = 'mp3';
+        
+        $this->audioFormat = $audioFormat; 
+    }
+
+    /**
+     * Sets the sample rate in hz of the audio output
+     *
+     * @param integer $sampleRate
+     * @return void
+     */
+    public function setSampleRate(int $sampleRate)
+    {
+        if (!in_array($sampleRate, [8000, 11025, 16000, 22050, 32000, 44100, 48000])) $sampleRate = 48000;
+        
+        $this->sampleRate = $sampleRate; 
+    }
+
+    /**
+     * Constructs the data required to be sent to the API and calls sendRequest()
+     * Returns an object containing the audio URL and various other data.
+     *
+     * @param string $voice
+     * @param string $text
+     * @return object
+     */
+    public function requestTTS(string $voice, string $text): object
+    {
+        // CereProc name their files very similarly to this library, so we can know the resulting URL of the audio
+        // before it's even generated. Unclear what the '1' refers to but likely to be account ID - 1 being their own demo.
+        // $cereprocFileUrl = 'https://cerevoice.s3.amazonaws.com/' . $voice . $this->sampleRate . '1' . md5($text) . '.' . $this->audioFormat;
+        
+        // Craft a request that makes it look as if it's come from CereProc's Drupal website
+        // First, a cookieKey is required. Seems like it can be any string, as long as the value sent in the headers matches the one sent with the payload
+        // Their JavaScript creates one using the following code: Math.random().toString(36).substr(2)
+        // Create a similar string...
+        mt_srand(date('Ymd'));
+        $cookieKey = base_convert(mt_rand(), 10, 16);
+
+        // XML to send as the payload
+        $payload = '<speakExtended key=\'' . $cookieKey . '\'><voice>' . $voice . '</voice><text>' . $text . '</text><audioFormat>' . $this->audioFormat . '</audioFormat></speakExtended>';
```

Comment:
Yes it would however I think this is copied and pasted from the old file and I simply forgot to replace the single quotes with double quotes for that XML attribute, so I'll just change that instead.